src/packet/ipv4.rs

```diff
@@ -76,11 +79,28 @@ pub struct Ipv4Option {
     copied: u1,
     class: u2,
     number: u5,
-    length: u8,
+    #[length_fn = "ipv4_options_length_length"]
+    length: Vec<u8>,
+    #[length_fn = "ipv4_options_payload_length"]
     #[payload]
     data: Vec<u8>,
 }
 
+fn ipv4_options_length_length(option: &Ipv4OptionPacket) -> usize {
+    match option.get_number() {
+        0 => 0,
```

Comment:
I would rather constants be used here, rather than integers - could you add something like `pnet::packet::ip::IpNextHeaderProtocols` and `pnet::packet::ethernet::EtherTypes`, and use constants defined in there instead?

The list doesn't need to be complete - I would like to avoid magic numbers though if possible.